### ISSUE TITLE
Starting a second instance on the same host doesn't throw an error or exit

### ISSUE THREAD

**Issue by voipmeister** [NuSkooler/enigma-bbs#547]
**Describe the Bug**
Not quite a bug (but no feature either I think), but while checking different versions of Node and Enigma I accidentally started a second instance. The second instance wasn't bound to port 8888/tcp ofcourse, but there was no warning message either. Would be nice to have a warning message and exit IMHO.

**To Reproduce**
Any relevant steps to reproduce the behavior:
Deploy a second installation of Enigma on the same host; start the first instance and the start the second instance.

**Desired Behavior**
Would be nice to have a warning message and exit IMHO.

**Actual Behavior**
The second instance isn't bound to port 8888/tcp and there's no warning message either.

**Environment**
* [x] I am using Node.js v14.x LTS or higher
* [x] `npm install` or `yarn` reports success
* Actual Node.js version (`node --version`): `v18.20.4`
* Operating system (`uname -a` on *nix systems): `Linux bbs01 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux` (Raspberry Pi 3B)
* Revision (`git rev-parse --short HEAD`): `9c0bc88f`
* Any additional relevant information about your setup: None, I think :)

**Comment by NuSkooler** [comment]
@voipmeister There should be errors in your log about binding to ports already in use. Enigma doesn't just panic and exit in this case, however.

**Comment by voipmeister** [comment]
Understood. Would it be possible to provide a little feedback on the console though? Granted, in production you'd not run into this often or easily, but I'm testing with various installs on the same VM and well... You know :)

**Comment by NuSkooler** [comment]
Fixed in #731, on `master` as edcad016.

@voipmeister — you were right, and it was a bit worse than "no warning message". Node's `server.listen(port, host, cb)` callback is a one-shot `'listening'` listener: it never receives an error and never fires at all on a failed bind. Every server here treated it as error-first, so `EADDRINUSE` left the startup sequence waiting on a callback that would never arrive — which is why the second instance neither complained nor exited nor finished starting. Telnet *did* notice the event, but logged it at `info` to the rotating log file, and the default logging config has no console stream, so nothing reached your terminal.

Starting a second instance now gives you:

```
Starting servers...
  ssh       FAILED: *:8889 is already in use - is another ENiGMA½ instance or another service already running?
```

and the process exits `1`.

Also in, from your follow-up about console feedback: startup now prints a line per bound server, so you can see what actually came up.

```
Starting servers...
  ssh       listening on *:8889
  telnet    listening on *:8888
  mrc       listening on *:5000
  http      listening on *:8080
System started!
```

Colour is used only when stdout is a TTY and [`NO_COLOR`](https://no-color.org/) is unset, so piped output and journals stay clean.

One upgrade note, also in [UPGRADE.md](https://github.com/NuSkooler/enigma-bbs/blob/master/UPGRADE.md): bind failures are now fatal for **every** server. NNTP previously logged a warning and carried on with the service silently absent. If you run NNTP, Gopher, or the web server on a port something else on the host also uses, a start that previously "worked" minus that service will now stop with an explicit error.

Thanks for the report — and for pushing on the console feedback, which is what made the fix actually useful rather than just correct.
